Edit the Resource Request guide

ptgott opened the pull request:
Closes #33417

Focus the guide more on its stated topic by moving tangentially related reference information at the end of the guide to more appropriate pages.

- Replace the `search_as_roles` section with a link. Much of the "Next steps" section is a long list of examples of Teleport roles that an admin could assign to `search_as_roles`. This is unnecessary, since a user familiar with a Teleport role won't need this information to understand `search_as_roles`. It is not the job of the Resource Requests guide to teach users about Teleport roles.
- Condense the "Next steps" guidance. There are "Next steps" sections. Make them one. Also turn the section re: Access Request plugins into a bullet.
- Move instructions for searching for and requesting Kubernetes resources into the tsh guide. These instructions are end users, so move them to the appropriate guide to make them more discoverable.

ptgott committed:
Edit the Resource Request guide

Closes #33417

Focus the guide more on its stated topic by moving tangentially related
reference information at the end of the guide to more appropriate pages.

- Replace the `search_as_roles` section with a link. Much of the "Next
  steps" section is a long list of examples of Teleport roles that an
  admin could assign to `search_as_roles`. This is unnecessary, since a
  user familiar with a Teleport role won't need this information to
  understand `search_as_roles`. It is not the job of the Resource Requests
  guide to teach users about Teleport roles.
- Condense the "Next steps" guidance. There are "Next steps" sections.
  Make them one. Also turn the section re: Access Request plugins into a
  bullet.
- Create a "Request Access to Resources" guide that shows end-users how
  to create Access Requests. Instead of manually adding an entry to the
  User Guides section in `sidebar.json`, this change configures the
  sidebar to generate entries for this section automatically.
- Add a subsection to the Kubernetes RBAC guide explaining implications
  for Access Requests and Kubernetes cluster resources. This information
  was removed from the docs when we cleaned up the Resource Requests
  page.

zmb3 commented on docs/pages/connect-your-client/tsh.mdx:
````diff
@@ -1126,6 +1127,139 @@ $ tsh ssh node1.leaf1
 $ ssh node1.leaf1
 ```
 
+## Request access to resources
````
This feels more like it belongs in an "access requests" section of the docs, not in a tsh section of the docs.

ravicious commented on docs/pages/admin-guides/access-controls/access-requests/resource-requests.mdx:
````diff
@@ -291,387 +287,15 @@ Getting updated certificates...
-  deny: {}
-```
-
-##### Preventing unintended access to Kubernetes resources
-
-If you are setting up a Teleport role to enable just-in-time access to a
-specific Kubernetes resources, you should set the role's `kubernetes_groups` and
-`kubernetes_users` to a role that has no access to Kubernetes resource beside
-the Kubernetes resources that Teleport is able to restrict access for.
-
-This is because, if a user requests access to a Kubernetes pod, and the request
-is approved, the Teleport Kubernetes Service will use the `kubernetes_groups`
-and `kubernetes_users` fields in the role to add impersonation headers to the user's
-requests to a Kubernetes API server. Under these conditions, Teleport will be able
-to restrict access to all Kubernetes resources kinds mentioned above except for
-the desired `pod`. Teleport is also able to restrict access to namespaced-scoped
-custom resources but not cluster-scoped custom resources - CRDs resources that
-are cluster-scoped will be accessible to the user if the principals in the
-`kubernetes_users` and `kubernetes_groups` fields have access to them.
-
-Requesting access to a Kubernetes Namespace allows you to access all resources
-in that namespace but you won't be able to access any other supported resources
-in the cluster.
````
This section was removed rather than moved but it seems important, no? I can see how the `node` and `kube_cluster` sections were unnecessary, but is the information contained within this section expressed anywhere else in the docs?

ptgott replied:
Addressed in e00f238043fe14e47775ecadb07909e1ab9c7cbf

zmb3 commented on docs/pages/connect-your-client/request-access.mdx:
This page is solely focused on Kubernetes access now. Should it be in the kube section of the docs?

ptgott replied:
I've made this more of a general-purpose Access Request guide for end users in https://github.com/gravitational/teleport/pull/52393/commits/5c36b6521ca53de99d6c296e830fbe145d38815c

ptgott commented:
TODO: I need to incorporate the edits made in this PR if we merge it before mine:
- https://github.com/gravitational/teleport/pull/55758

creack commented on docs/pages/connect-your-client/request-access.mdx:
````diff
@@ -0,0 +1,174 @@
+   command:
+
+   ```code
+   $ tsh request create --resource /teleport.example.com/pod/mycluster/*/^nginx-[a-z0-9-]+$
````
nit: the resource id should be quoted to avoid * , [] and $ to be evaluated by the shell.

ptgott commented:
@creack @zmb3 @ravicious I've reworked this PR to incorporate the changes made in #55758

ptgott committed:
Respond to feedback

- Indicate that Teleport Enterprise is required to create Access
  Requests in the Web UI.
- Mention Teleport Connect.
- Mention `kube_resource`.
